pkg/volume/volume.go

```diff
@@ -131,10 +131,8 @@ type Deleter interface {
 
 // Attacher can attach a volume to a node.
 type Attacher interface {
-	Volume
-
 	// Attach the volume specified by the given spec to the given host
-	Attach(spec *Spec, hostName string) error
+	Attach(host VolumeHost, spec *Spec, hostName string) error
```

Comment:
Hmm.. right now the Attacher and Detacher have no reference to the plugin. I did that on purpose to keep the dependencies as low in the hierarchy as possible. In order to do this I would need to add that reference; either in the constructor or pass it directly to the function. WDYT ?